packages/next/pages/_error.tsx

```diff
@@ -49,9 +50,10 @@ export default class Error<P = {}> extends React.Component<P & ErrorProps> {
           </title>
         </Head>
         <div>
-          <style
-            dangerouslySetInnerHTML={{
-              __html: `
+          {withDarkMode ? (
+            <style
+              dangerouslySetInnerHTML={{
+                __html: `
                 body { margin: 0; color: #000; background: #fff; }
                 .next-error-h1 {
                   border-right: 1px solid rgba(0, 0, 0, .3);
```

Comment:
If this is strictly disabling the dark mode color scheme could we conditionally add the `@media` content to this style template literal instead? That ensures we maintain the previous styling without a breaking change E.g. 
```jsx
<style
  dangerouslySetInnerHTML={{
    __html: `
    body { margin: 0; color: #000; background: #fff; }
    .next-error-h1 {
      border-right: 1px solid rgba(0, 0, 0, .3);
    }
    ${withDarkMode ? `@media (prefers-color-scheme: dark) {
      body { color: #fff; background: #000; }
      .next-error-h1 {
        border-right: 1px solid rgba(255, 255, 255, .3);
      }
    }` : ``}`,
  }}
/>
```

Reply:
This is where things get complicated I guess.

The real aim of the PR, is to have an ability to go back to the old style of the `Error` page, that did not set any colors, only `margin: 0;` on body.

Perhaps I've chosen a bad name. `withColorSchemePreference` is a bit long, but more accurate wouldn't you say?

Just disabling the media query for color preference, still forces the page to have `#fff`, and color `#000`, which might not really be what users have in their stylesheets.

Reply:
The error page has always set the `background` and `color` values to ensure global CSS doesn't conflict with styles defined here. If more customization is desired then it sounds like a completely separate error page should be used instead, related documentation here https://nextjs.org/docs/advanced-features/custom-error-page#404-page

Reply:
Right, but then we could settle on your suggestion, to selectively opt-out form the colour preference for dark mode.